cookbooks/bcpc-hadoop/templates/default/hdp_yarn-site.xml.erb

```diff
@@ -303,4 +303,10 @@
     <name>yarn.scheduler.fair.preemption</name>
     <value><%= node["bcpc"]["hadoop"]["yarn"]["scheduler"]["fair"]["preemption"] %></value>
   </property>
+
+  <property>
+    <name>yarn.log.server.url</name>
+    <value>http://<%="#{float_host(@hs_hosts.map{|i| i[:hostname] }.sort.first)}:19888" if not @hs_hosts.empty?%>/jobhistory/logs</value>
```

Comment:
Agree with @bijugs better off just making this a conditional.  Or don;t ask, if this is empty you have a whole new set of problems.